Based on the new version of Aliyun CLI

bd233 opened the pull request.

bd233 commented:
> Can one of the admins verify this patch?

yes, they can.

oalbrigt commented:
You'll have to add some logic that detects whether the old or new version is present, and use correct command/parameters based on that.

bd233 commented:
> 您将必须添加一些逻辑以检测是否存在旧版本或新版本，并基于此使用正确的命令/参数。

The older version of the cli tool aliyun is no longer maintained and should be installed manually by the user before use this agent. Therefore, it does not happen at the same time as both the old version and the new version.

We will update its operation instructions on aliyun official website and SUSE official website.

oalbrigt commented:
RHEL uses the older version, and you cant expect all versions of all distros to be upgraded to the newest version of the CLI tool.

oalbrigt commented:
I will cherry-pick your commit, and add the logic to detect if the old or new version is available.

Should `region_id` and `endpoint` be available as parameters to set by users (if so we'll need to add it to the agents metadata)?

bd233 commented:
> I will cherry-pick your commit, and add the logic to detect if the old or new version is available.
> 
> Should `region_id` and `endpoint` be available as parameters to set by users (if so we'll need to add it to the agents metadata)?
Yes, I have added the "aliyuncli path" and "endpoint" parameters. I have modified them to support the new version and the old version. I will submit them later.

oalbrigt commented on heartbeat/aliyun-vpc-move-ip:
```diff
@@ -37,27 +54,37 @@ USAGE="usage: $0 {start|stop|status|meta-data}";
+	$cmd
+}
+
+request_delete_rote_entry() {
```
You got a typo here in "route" (and where it's called below)

bd233 replied:
you are right，tks

oalbrigt commented:
ok to test

oalbrigt commented:
Looks good to me. Thanks.

I'll merge and add some auto-detection so it doesnt fail for users when they upgrade.

oalbrigt commented:
I've added auto-detection in https://github.com/ClusterLabs/resource-agents/pull/1420.

Feel free to test it, and tell me if there are any issues.
